benchmarking/smartseq2/picard_qc_analysis/QC_metrics_analysis.R

```diff
@@ -0,0 +1,387 @@
+        size = 18,
+        face = 'bold'
+      ),
+      axis.text.x = element_text(
+        color = 'black',
+        size = 18,
+        face = 'bold'
+      )
+    )
+  p <-
+    p + theme(legend.text = element_text(size = 15, face = 'bold'),
+              legend.title = element_blank())
+  return(p)
+}
+# lm test
+# scatter plot and fitted line
+# also plot 1x1 line
+plotlm <- function(df) {
+  # lm test
+  x <- df$Base
+  y <- df$Updated
+  fit <- lm(y ~ x)
+  # summary
+  sfit <- summary(fit)
+  # r2
+  r2  <- round(sfit$adj.r.squared, 3)
+  # intercept and slope
+  beta <- round(sfit$coefficients[2], 3)
+  a <- round(sfit$coefficients[1], 3)
+  # f-stats and p values
+  f <- sfit$fstatistic
+  fpval <- pf(f[1], f[2], f[3], lower.tail = F)
+  fpval.s <- convert2star(fpval)
+  # start to plot
+  coefs <-
+    data.frame(
+      'ic' = c(0, a),
+      's' = c(1, beta),
+      'tl' = c('1x1', 'fitted')
+    )
+  my.formula <- y ~ x
+  p <-
+    ggplot(data = df, aes(x = Base, y = Updated)) + geom_point(shape = 1) + theme_bw(base_size = 20)
+  p <- p + stat_poly_eq(
+    formula = my.formula,
+    eq.with.lhs = "italic(hat(y))~`=`~",
+    aes(
+      label = paste(..eq.label.., ..rr.label.., sep = "~~~"),
+      size = 8
+    ),
+    parse = TRUE,
+    size = 8
+  )
+  p <-
+    p + stat_fit_glance(
+      method = 'lm',
+      method.args = list(formula = my.formula),
+      geom = 'text',
+      aes(
+        label = paste("P-value:", fpval.s , sep = ""),
+        size = 12
+      ),
+      label.x.npc = 'left',
+      label.y.npc = 0.85,
+      size = 8
+    )
+  p <- p + theme(legend.position = "top")
+  p <- addTheme(p)
+  p <- p + xlab(paste('Base')) + ylab(paste('Updated'))
+  p <-
+    p + geom_abline(
+      data = coefs,
+      mapping = aes(
+        slope = s,
+        intercept = ic,
+        linetype = factor(tl),
+        color = factor(tl)
+      )
+    )
+  return(list(
+      'p' = p,
+      'beta' = beta,
+      'a' = a,
+      'r2' = r2,
+      'fpval' = fpval
+    ))
+}
+# plot violin plot 
+# run mean comparison test
+plotBox <- function(df) {
+  mz <- melt(df)
+  p <-
+    ggviolin(
+      mz,
+      x = 'variable',
+      y = 'value',
+      color = "variable",
+      palette = c("#00AFBB", "#E7B800"),
+      fill = "variable",
+      add = "boxplot",
+      add.params = list(fill = "white")
+    )
+  p <-
+    p + stat_compare_means(comparisons = list(c('Base', 'Updated')), label = "p.signif") +
+    stat_compare_means() + # Add pairwise comparisons p-value
+    grids(linetype = "dashed")
+  p <- addTheme(p)
+  stat <- wilcox.test(x, y)
+  return(list('p' = p, 'pval' = stat$p.value))
+}
+# plot histogram/density plot
+plothist <- function(df) {
+  # melt dataframe
+  mz <- melt(df)
+  # group mean values
+  mu <- apply(df, 2, mean)
+  density.p <- ggdensity(
+    mz,
+    x = "value",
+    add = "mean",
+    rug = TRUE,
+    color = "variable",
+    fill = "variable",
+    palette = c("#00AFBB", "#E7B800")
```

Comment:
If you are using these colors in multiple plots, you could potentially extract them for readability. Actually, can palette be part of your themes you've already extracted, or does that not work?

Reply:
I guess two ways to do this,  one  you can define a *color* or  just set up as default color somewhere in the R script

Reply:
I just set up palette at the beginning of script  and all plots have the same color theme now.